integration-tests/src/main/java/org/apache/druid/testing/utils/QueryResultVerifier.java

```diff
@@ -41,19 +47,35 @@ public static boolean compareResults(
       if (fieldsToTest != null && !fieldsToTest.isEmpty()) {
         for (String field : fieldsToTest) {
           if (!actualRes.get(field).equals(expRes.get(field))) {
-            return false;
+            String mismatchMessage = StringUtils.format(
+                "Field [%s] mismatch. Expected: [%s], Actual: [%s]",
```

Comment:
Something like this would be more informative:
```suggestion
                "Mismatch in row [%d], col [%s]. Expected: [%s], Actual: [%s]",
```